veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,345 @@
+
+Since virt-launcher drops the root user, this method is secure. However, KubeVirt follows Kubernetes' `Restricted` Pod Security Standard, which disallows CAP_NET_ADMIN.
+
+## Scalability
+
+- `passt` has a memory overhead of ~250Mi per VM. Users running VMs at scale should revert to other bindings if memory overhead is a concern.
+-  VMs with a large number of TCP connections will take longer to live migrate. That's because sockets are handled in a serial
+ fashion, each one separately sent back and forth to passt-repair for handling.
+- The number of concurrent live migration per node is already limited and bound by the number of virt-handler worker goroutines.
+
+## Update/Rollback Compatibility
+
+- Live migration of VMs with `passt` network binding, from nodes running older versions of KubeVirt to an upgraded KubeVirt environment will result in TCP connections being reset, and no `vhost-user` performance boost. Subsequent migrations will retain the same limitations.
+- From the client perspective, the connection would be reset and a new handshake will be necessary.      
+- Rollback of passt migration functionality (in case system stability is impacted) will be possible by disabling of the feature gate. It is expected that passt would still work, but live migrations would disconnect TCP connections.
+
+### Recommended upgrade procedure
+Once the cluster is fully upgraded to KubeVirt 1.6 release, VMs running older versions of `passt` network binding, should be restarted.
+
+It is recommended to upgrade the cluster in 2 separate steps:
+1. A full KubeVirt upgrade, without bumping the `passt` sidecar image tag in the cluster-wide configuration.
+2. Only once everything else has been upgraded. Bump the sidecar image tag in the configuration.
+Once the entire system and configuration is updated, `passt` based VMs should be restarted to benefit from new `passt` features. If they're not restarted, they are expected to continue to work as before.
+As VMs are restarted, they will be using the new virt-launcher image containing the new passt/libvirt stack. The updated cluster config will use the new `passt` sidecar image which will configure the DomainXML to use `vhost-user`.
+By the time that live migration is requested, the new virt-handler will run `passt-repair` and migration will be seamless. 
+
+> [!NOTE]
+> While any of the components mentioned in the solution are not fully upgraded, seamless migration is not possible, and TCP sessions can be expected to disconnect during migration.
+
+> [!WARNING]
+> Do not run KubeVirt version 1.5 or older with passt network binding plugin of version 1.6, as the plugin is not backward compatible and is not expected to work.
```

Comment:
maybe worth to add assertion in code for that ? (and then either to mentioned it here or just to know about it)